### PR TITLE
bug 1406546: potential fix for kuma test reliability issue in jenkins

### PR DESCRIPTION
**Pull request by escattone**
This PR addresses https://github.com/mozmeao/infra/issues/650 and https://github.com/mozmeao/infra/issues/602, as well as https://github.com/mozmeao/infra/issues/595.

* use `docker-compose down` to ensure a clean context both before and after the Kuma tests (I'm hoping this addresses the symptoms of https://github.com/mozmeao/infra/issues/602)
* notify `#mdndev` in IRC on failure

I also added IRC notification on failure to the `default` case, but I would be happy to remove that if we'd rather not clutter `#mdndev` with failure notifications for random branches.

### CODE REVIEW

File: Jenkinsfiles/utils.groovy [hunk 1]
```diff
@@ -55,34 +55,38 @@ def notify_irc(Map args) {
     sh command
 }
 
-def make(cmd, cmd_display) {
-    def target = get_target()
-    def tag = get_commit_tag()
-    def nick = "mdn${target}push"
-    def repo_upper = get_repo_name().toUpperCase()
+def sh_with_notify(cmd, display, notify_on_success=false) {
+    def nick = "mdn-${env.BRANCH_NAME}"
```
Comment (metadave) [inline Jenkinsfiles/utils.groovy]: does the nick change with every branch on purpose?

Comment (escattone) [reply Jenkinsfiles/utils.groovy]: Yes, I thought it'd be a quick indicator of where the message is coming from (although the branch name is also in the message). It's helpful for example when we're doing a push of both kuma and kumascript at the same time, since they get different nicks.

I'm totally open to a different approach though!

Comment (jwhitlock) [reply Jenkinsfiles/utils.groovy]: It does make my IRC a bit wordy:

screenshot 2017-11-16 16 10 14

Let's keep this for a week or so and a few pushes. I might have a stronger opinion then.
